src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -259,43 +363,114 @@ def _construct_job_output_state_set(
+                )
 
         return output_states
 
-    def get_job_states(self, job_ids: List[str]) -> dict:
-        job_ids, error_ids = self._check_job_list(job_ids)
+    def get_job_states(self, job_ids: List[str], ts: int = None) -> dict:
+        """
+        Retrieves the job states for the supplied job_ids, with the option to
+        replace any jobs that have not been updated since ts with a short stub
+
+        (Omitting some error states from job.output_state()) the output is generally like:
+        {
+            "job_id_0": {  # dict generated by job.output_state()
+                "job_id": "job_id_0":
+                "jobState": {  # modified job state from EE2
+                    ...
+                },
+                "outputWidgetInfo": {
+                    ...
+                }
+            },
+            "job_id_1": {  # dict generated by job.output_state()
+                "job_id": "job_id_1":
+                "jobState": {  # modified job state from EE2
+                    ...
+                },
+                "outputWidgetInfo": None
+            },
+            ...,
+            "ee2_error_id_0": {  # dict generated by job.output_state() with EE2 error message added
+                "job_id": "ee2_error_id_0":
+                "jobState": {  # modified job state from EE2
+                    ...
+                },
+                "outputWidgetInfo": ...,
+                "error": <EE2 error message>
+            },
+            ...,
+            "nonexistent_error_id_0": {  # jobs that cannot be found in the `_running_jobs` index
+                "job_id": "nonexistent_error_id_0",
+                "error": "Cannot find job with ID nonexistent_error_id_0"
+            },
+            ...
+            "not_updated_id_0": {  # jobs that have not been updated since ts
+                "job_id": "not_updated_id_0",
+                "error": "Job with ID not_updated_id_0 has not been updated since ts 123085709827"
+            }
+        }
+
+        :param job_ids: job IDs to retrieve job state data for
+        :type job_ids: List[str]
+        :param ts: timestamp (as generated by time.time_ns()) to filter the jobs, defaults to None
+        :type ts: int, optional
+
+        :return: dictionary of job states, indexed by job ID
+        :rtype: dict
+        """
+        job_ids, not_found_ids = self._check_job_list(job_ids)
         output_states = self._construct_job_output_state_set(job_ids)
-        return self.add_errors_to_results(output_states, error_ids)
+        not_updated_ids = []
+        if ts is not None:
+            for job_id in output_states:
+                if self.get_job(job_id).last_updated <= ts:
+                    not_updated_ids.append(job_id)
+        self.add_errors_to_results(output_states, not_found_ids, OutputStateErrMsg.NOT_FOUND)
+        self.add_errors_to_results(output_states, not_updated_ids, OutputStateErrMsg.NOT_UPDATED, ts)
```

Comment:
Unfortunately this is not what I asked for in the Jira ticket -- if a job hasn't been updated, the BE is supposed just to omit it from the results it returns.

Once the docs PR is merged and this PR is rebased on develop, I'll set this up locally and do some experimenting.

Reply:
So the description was a little more open-ended and inconsistent with the acceptance critera. So I interpreted the entire JIRA ticket as being open-ended.

Reply:
Sorry, I think I got completely tripped up by this part of the ticket
```
don’t include the jobState dict in the job output, just return job_id and last_checked for each job in the request?
```